### PR TITLE
Refactor survey task thumbnail options

### PR DESCRIPTION
**Pull request by mcbouslog**
Staging branch URL: https://pr-6669.pfe-preview.zooniverse.org

Towards - https://github.com/zooniverse/front-end-monorepo/issues/3120
FEM PR - https://github.com/zooniverse/front-end-monorepo/pull/4841

- survey task default/dynamic thumbnail size - https://pr-6669.pfe-preview.zooniverse.org/lab/1634/workflows/3765?env=staging&pfeLab=true
- survey task always show (old) - https://pr-6669.pfe-preview.zooniverse.org/lab/1634/workflows/3729?env=staging&pfeLab=true
- survey task always show (new) - https://pr-6669.pfe-preview.zooniverse.org/lab/1634/workflows/3764?env=staging&pfeLab=true
- survey task always hide (new)- https://pr-6669.pfe-preview.zooniverse.org/lab/1634/workflows/3763?env=staging&pfeLab=true

Describe your changes.
- add survey task `thumbnails` property
- refactor survey task editor to support setting `thumbnails` property to `"show"` (replacing `alwaysShowThumbnails` boolean property), `"hide"`, or `"default"` (maybe this should be something else, like "dynamic", or open to suggestion)
- refactor Choices component for `thumbnails` property to support:
  - (old/default) dynamically display thumbnail size per choices count
  - (old) legacy `alwaysShowThumbnails` property to always show thumbnails
  - (new) always show thumbnails
  - (new) always hide thumbnails

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `npm ci` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on GitHub Actions?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

File: app/classifier/tasks/survey/chooser/Choices.jsx [hunk 1]
```diff
@@ -45,9 +45,16 @@ class Choices extends React.Component {
   }
 
   whatSizeThumbnails({ length }) {
-    if (this.props.task.alwaysShowThumbnails) {
+    // if new survey task thumbnails property is undefined and legacy alwaysShowThumbnails is true, then show thumbnails to support legacy alwaysShowThumbnails functionality
```
Comment (goplayoutside3) [inline app/classifier/tasks/survey/chooser/Choices.jsx:48]:
```suggestion
    // if new survey task.thumbnails property is undefined and legacy task.alwaysShowThumbnails is true, then show thumbnails to support legacy alwaysShowThumbnails functionality
```

Suggestion for clarity

File: app/classifier/tasks/survey/editor.cjsx [hunk 2]
```diff
@@ -305,17 +305,35 @@ module.exports = createReactClass
...
+            </small>
+          </p>
+          <p>
+            <small>
+              <strong>Show</strong> - will always show thumbnails as small, regardless of the number of choices shown.
+            </small>
+          </p>
+          <p>
+            <small>
+              <strong>Hide</strong> - will never show thumbnails.
```
Comment (goplayoutside3) [inline app/classifier/tasks/survey/editor.cjsx:324]:
```suggestion
              <strong>Hide All</strong> - will never show thumbnails.
```

File: app/classifier/tasks/survey/editor.cjsx [hunk 3]
```diff
@@ -305,17 +305,35 @@ module.exports = createReactClass
...
-            If checked, then thumbnails will always show as small. If unchecked, then thumbnails will show as small, medium, large, or not at all (when choices > 30) based on the number of choices shown. The number of choices shown will change based on filters.
-          </small>
-        </p>
+        <AutoSave resource={@props.workflow}>
+          <label htmlFor="#{@props.taskPrefix}.thumbnails">
+              <span className="form-label">Thumbnails</span>
+          </label>
+          <p>
+            <small>
+              <strong>Default</strong> - will show thumbnails as small, medium, large, or not at all (when choices > 30) based on the number of choices shown. The number of choices shown will change based on filters.
```
Comment (goplayoutside3) [inline app/classifier/tasks/survey/editor.cjsx:314]:
```suggestion
              <strong>Default</strong> - will show thumbnails as small, medium, large, or not at all (when choices > 30) based on the number of choices shown. Note that volunteer-selected filters change the number of choices shown.
```

**Review by goplayoutside3 — APPROVED** [review]
Looks good! I think "Default" is a good label for the dynamic behavior because it suggests that the UI will handle everything smoothly.

File: app/classifier/tasks/survey/editor.cjsx [hunk 4]
```diff
@@ -305,17 +305,35 @@ module.exports = createReactClass
...
+            </small>
+          </p>
+          <p>
+            <small>
+              <strong>Show</strong> - will always show thumbnails as small, regardless of the number of choices shown.
```
[the remark this answers is not in the document]
Comment (mcbouslog) [reply app/classifier/tasks/survey/editor.cjsx:319]: Maybe the related select option value should be changed from `"show"` to `"show small"`?

**Pull request merged**